src/Wallabag/ApiBundle/Controller/DeveloperController.php

```diff
@@ -20,9 +22,62 @@ class DeveloperController extends Controller
     public function indexAction()
     {
         $clients = $this->getDoctrine()->getRepository('WallabagApiBundle:Client')->findByUser($this->getUser()->getId());
+        $globalClients = $this->getDoctrine()->getRepository('WallabagApiBundle:Client')->findByUser(null);
 
         return $this->render('@WallabagCore/themes/common/Developer/index.html.twig', [
             'clients' => $clients,
+            'global_clients' => $globalClients,
+        ]);
+    }
+
+    /**
+     * Create a global client (an app) for all users
+     *
+     * @param Request $request
+     *
+     * @Route("/developer/client/global/create", name="developer_create_global_client")
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function createGlobalClientAction(Request $request)
+    {
+        $em = $this->getDoctrine()->getManager();
+        $client = new Client();
+        $clientForm = $this->createForm(GlobalClientType::class, $client);
+        $clientForm->handleRequest($request);
+
+        if ($clientForm->isSubmitted() && $clientForm->isValid()) {
+
+            /** Handling the application icon */
+            $file = $client->getImage();
+            $fileName = md5(uniqid('', true)).'.'.$file->guessExtension();
+
+            $file->move(
+                $this->getParameter('wallabag_api.applications_icon_path'),
+                $fileName
+            );
+
+            $client->setImage($fileName);
+
+
+            $client->setAllowedGrantTypes(['token', 'authorization_code', 'refresh_token']);
+            $em->persist($client);
+            $em->flush();
+
+            $this->get('session')->getFlashBag()->add(
+                'notice',
+                $this->get('translator')->trans('flashes.developer.notice.client_created', ['%name%' => $client->getName()])
+            );
+
+            return $this->render('@WallabagCore/themes/common/Developer/client_parameters.html.twig', [
```

Comment:
Maybe you can display the uploaded picture when we just have created the client.